src/sharedTypes.ts

```diff
@@ -1,9 +1,14 @@
 import { NativeStackScreenProps } from "@react-navigation/native-stack";
 import { MessageDescriptor } from "react-intl";
 
-import { SyncScreens } from "./navigation/NavigationContainer";
+import { Screens } from "./navigation/NavigationContainer";
 
-export type SyncScreenComponent<ScreenName extends keyof SyncScreens> =
-  React.FC<NativeStackScreenProps<SyncScreens, ScreenName>> & {
-    navTitle: MessageDescriptor;
-  };
+export type ScreenComponent<ScreenName extends keyof Screens> = React.FC<
+  NativeStackScreenProps<Screens, ScreenName>
+> &
+  // Home screen doesn't have a nav header
+  (ScreenName extends "Home"
+    ? {}
```

Comment:
there's probably a better way of doing this but got a bit lazy and went with this since it works fine 🤷‍♂️

Reply:
There are less clunkier ways, but this is the most type safe way that I can think of.